spacy/strings.pyi

```diff
@@ -1,21 +1,42 @@
-from typing import Optional, Iterable, Iterator, Union, Any, Tuple
+from typing import Optional, Iterable, Iterator, Union, Any, Tuple, overload
 from pathlib import Path
 
 class StringStore:
     def __init__(self, strings: Optional[Iterable[str]]) -> None: ...
-    def __getitem__(self, string_or_hash: Union[str, int]) -> Union[str, int]: ...
-    def as_int(self, key: Union[str, int]) -> int: ...
-    def as_string(self, string_or_hash: Union[str, int]) -> str: ...
+    @overload
+    def __getitem__(self, string_or_hash: str) -> int: ...
+    @overload
+    def __getitem__(self, string_or_hash: int) -> str: ...
+    @overload
+    def as_int(self, string_or_hash: str) -> int: ...
+    @overload
+    def as_int(self, string_or_hash: int) -> int: ...
+    @overload
+    def as_string(self, string_or_hash: str) -> str: ...
+    @overload
+    def as_string(self, string_or_hash: int) -> str: ...
     def add(self, string: str) -> int: ...
     def items(self) -> Tuple[int, str]: ...
     def __len__(self) -> int: ...
-    def __contains__(self, string_or_hash: Union[str, int]) -> bool: ...
+    @overload
+    def __contains__(self, string_or_hash: str) -> bool: ...
+    @overload
+    def __contains__(self, string_or_hash: int) -> bool: ...
```

Comment:
This could stay a `Union`

Reply:
For future reference, what criteria do we use to select which functions need `@overload` annotations? Do we only wnant to overload when the return type is dependent on the input type?

Reply:
It's if the input type determines the output type, not just if there's a `Union`.

Reply:
There's no benefit to not using a `Union` above.